File: Low Level Design (LLD)/Problems/06. Splitwise/Java (with Springboot)/splitwise/src/main/java/com/splitwise/splitwise/model/ExactExpense.java
package com.splitwise.splitwise.model;

import com.splitwise.splitwise.model.enums.ExpenseType;

import java.util.*;

public class ExactExpense extends Expense {
    private List<Double> exactAmounts;

    public ExactExpense(String expenseId, String description, double amount, User paidBy, List<User> participants, List<Double> exactAmounts) {
        super(expenseId, description, amount, paidBy, participants, ExpenseType.EQUAL);
        this.exactAmounts = exactAmounts;
    }

    @Override
    public void calculateSplit() {
        for (int i = 0; i < participants.size(); i++) {
            userShare.put(participants.get(i), exactAmounts.get(i));
        }
    }
}

File: Low Level Design (LLD)/Problems/06. Splitwise/Cpp/model/src/ExactExpense.cpp
#include "../include/ExactExpense.h"

ExactExpense::ExactExpense(const std::string& expenseId,
                           const std::string& description,
                           double amount,
                           std::shared_ptr<User> paidBy,
                           const std::vector<std::shared_ptr<User>>& participants,
                           const std::vector<double>& exactAmounts)
    : Expense(expenseId, description, amount, paidBy, participants, ExpenseType::EXACT),
      exactAmounts(exactAmounts) {}

void ExactExpense::calculateSplit() {
    auto& userShareMap = getUserShareMutable();

    for (size_t i = 0; i < participants.size(); ++i) {
        userShareMap[participants[i]] = exactAmounts[i];
    }
}
